examples/ipfs-kad.rs

```diff
@@ -91,7 +91,7 @@ fn main() -> Result<(), Box<dyn Error>> {
     };
 
     println!("Searching for the closest peers to {:?}", to_search);
-    swarm.get_closest_peers(to_search);
+    swarm.get_closest_peers(to_search.to_bytes());
```

Comment:
No longer being able to pass `PeerId`s or `Multihash`es to the Kademlia API, as in this case to `get_closest_peers()`, kind of defeats the purpose of the generic `key: K` where `K: Borrow<[u8]>` argument. Could we change these arguments, which I think exist for `Kademlia::kbucket()` and `Kademlia::get_closest_peers()` to be `key: K` where `K: Into<kbucket::Key<K>>`?